web-module/src/main/webapp/resources/structEditJS/StructObjectListEdit.js

```diff
@@ -121,13 +124,23 @@
     return objNb;
   };
 
-  $j(document).ready(init_structObjectListEdit);
-  $j(document).ready(function() {
-    $(document.body).observe('celements:contentChanged', init_structObjectListEdit);
-    $(document.body).observe('tabedit:successfulSaved', function(event) {
-      location.reload();
-    });
-  });
-  
+  const reloadPage = () => _createObjectCount && location.reload();
+
+  const reloadOnSaveHandler = () => {
+    const structManager = window.celStructEditorManager;
+    if (structManager) {
+      if (structManager.isStartFinished()) {
+        structManager.celObserve('structEdit:saveAndContinueButtonSuccessful', reloadPage);
+      } else {
+        structManager.celObserve('structEdit:finishedLoading', reloadOnSaveHandler);
+      }
+    } else {
+      $(document.body).observe('tabedit:saveAndContinueButtonSuccessful', );
+    }
+  };
+
+  document.addEventListener('DOMContentLoaded', init_structObjectListEdit);
+  $(document.body).observe('celements:contentChanged', init_structObjectListEdit);
+  document.addEventListener('DOMContentLoaded', reloadOnSaveHandler);
```

Comment:
if I remember correctly `DOMContentLoaded` isn't a full replacement for `jquery.ready`. do you know if there is a crucial difference for this refactoring?

Reply:
jquery.ready is firing immediately if DOMContentLoaded is already over or else it is registering on DOMContentLoaded. DOMContentLoaded itself is only firing once. If you register to late on DOMContentLoaded, it will be never called.

Reply:
```
if (['complete', 'interactive'].includes(document.readyState)) {
  console.debug("initGoogleMaps in dynamic mode loaded immediately", document.readyState);
  //TODO your work directly
} else {
  console.debug("initGoogleMaps in sync mode loaded on defer ", document.readyState);
  document.addEventListener('DOMContentLoaded', () => registerYourMethod;
}

```